### ISSUE TITLE
Alternative to duti?

### ISSUE THREAD

**Issue by ssbarnea** [moretension/duti#46]
As duti does not really seems actively maintained, I am wondering if there is any alternative can that be used to repair messed file associations on macos? ... especially if it would include an option to disassociate files.

**Comment by deltabravozulu** [comment]
https://github.com/Lord-Kamina/SwiftDefaultApps

**Comment by faelin** [comment]
> https://github.com/Lord-Kamina/SwiftDefaultApps

This project is even less well maintained than duti (last commit in Oct 2020 vs 3 days ago for duti)

**Comment by ssbarnea** [comment]
To be honest I would be glad to contribute to a Python rewrite of duti. This kind of project does not really need speed, it needs to be easy to use and maintain.

**Comment by iloveitaly** [comment]
Is there something broken with the current implementation that you'd like
to fix with a python version? Not against it, but duti is very stable and
I'd opt for a simple tweak on the existing version vs a full rewrite.

**Comment by ssbarnea** [comment]
Yes there is. As a command line tool, duti fails to provide any help when installed. `--help` should list all available commands and also some examples of usage. Reading https://clig.dev/ could help (not affiliated with them).

If it would have a way to display help, it would probably be easy to contribute PRs to improve the help, but there is is none right now.

**Comment by faelin** [comment]
I would add that the `--help` response even *could* be written in Python, without changing the rest of the app virtually at all (except for reading the help flag and redirecting args to the Python).

Not a _good_ way to do it, but if someone really wants to contribute and can only do it on Python, then there are still ways to do it without rewriting the code base.

**Comment by faelin** [comment]
However, a test suite and/or added documentation would be a huge win for community maintenance!

**Comment by ssbarnea** [comment]
That means that we could make it a python tool, ease the way we install it while still keeping its c low level implementation. Is not a rewrite, is more of wrapping it so we can make the UI easier to maintain.

**Comment by tsonglew** [comment]
[dutis](https://github.com/tsonglew/dutis) is a wrapper of duti, providing an interactive way to set default applications in command line

**Comment by iloveitaly** [comment]
@faelin I have maintainer privileges and can approve any improvements/PRs here!

**Comment by b-hal** [comment]
> To be honest I would be glad to contribute to a Python rewrite of duti. This kind of project does not really need speed, it needs to be easy to use and maintain.

No, since [Apple removed Python from macOS](https://www.jamf.com/blog/python2-is-gone-apple-macos/ ), the best solution for a deployable (e.g. with a MDM) is a small binary like duti is now.

**Comment by philocalyst** [comment]
A python version can be found here: [repo](https://github.com/lkubb/dooti) -- but I am working on a fully-featured swift version on my own time as well, it should be complete by the end of the week!

**Comment by philocalyst** [comment]
My version is now live :) 
https://github.com/philocalyst/infat/tree/v0.1.0-alpha

**Comment by dnicolson** [comment]
This has been created recently: https://github.com/klabast/macOSDefaultApps
